interp/builtin.go

```diff
@@ -606,6 +604,10 @@ func (r *Runner) builtinCode(ctx context.Context, pos syntax.Pos, name string, a
 			r.setVarString(name, val)
 		}
 
+		if err != nil {
+			return 1
+		}
```

Comment:
> can you document why we check the error after using the returned value from readLine now?

In retrospect, it's not really clear, but I mentioned it in the PR comment: so that if you press ^D (EOF) when reading into a shell variable, it sets the variable to "".

I'll be happy to add a comment to the actual code, of course. This is just so you know.

Reply:
LGTM